tfjs-layers/src/layers/nlp/tokenizers_utils.ts

```diff
@@ -141,3 +141,164 @@ export class BytePairTokenizerCache {
     return arrKeys.map(key => this._cache.get(key));
   }
 }
+
+/**
+ * Remove certain strings from input tensor.
+ */
+export async function removeStringsFromInputs(
+  inputs: Tensor[], stringToRemove: string): Promise<Tensor[]> {
+
+  const stringArrInputs = await Promise.all(inputs.map(async input =>
+    (await input.data() as unknown as string[])));
+
+  const filteredStrArrays = stringArrInputs
+    .map(arr => arr.filter(s => s !== stringToRemove))
+    .filter(arr => arr.length > 0);
```

Comment:
It's surprising to me that it flattens in the inner dimension and also the outer dimension (i.e. removes empty arrays). Are you sure this is what the python version is doing? 

Specifically, see [this line](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/tokenizers/byte_pair_tokenizer.py#L97) that's computing the row lengths. It may result in a zero for a row length, but that row should still be included in the ragged tensor, right? e.g.
```
>>> tf.RaggedTensor.from_row_lengths(values=[1,2,3,4,5,6,7,8], row_lengths=[2,0,3,3])
<tf.RaggedTensor [[1, 2], [], [3, 4, 5], [6, 7, 8]]>
```